### PR TITLE
Add support for filtering on state name to `prefect flow-run ls`

### LINKED ISSUE #6065: missing arg LATE on cli
## Description
prefect==2.0b13
`LATE` is not possible as an option for `--state-type` param on the prefect cli.

## Reproduction / Example
```bash
❯ prefect flow-run ls --help
                                                                                                                   
 Usage: prefect flow-run ls [OPTIONS]                                                                              
                                                                                                                   
 View recent flow runs or flow runs for specific flows                                                             
                                                                                                                   
╭─ Options ───────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --flow-name         TEXT                                          [default: None]                               │
│ --limit             INTEGER                                       [default: 15]                                 │
│ --state-type        [SCHEDULED|PENDING|RUNNING|COMPLETED|FAILED|  [default: None]                               │
│                     CANCELLED|CRASHED]                                                                          │
│ --help                                                            Show this message and exit.                   │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

### PR DESCRIPTION
**Pull request by bunchesofdonald**
### Summary
Adds a `--state` parameter to `prefect flow-run ls` to filter flow runs by the name of the state. Closes #6065 

### Steps Taken to QA Changes
- Unittests and manual verification

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- No tests or issue needed
- [x] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [ ] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Comment by madkinsz** [comment]
I think just `--state` would be nice and follows our "name is the default filtering option". Users shouldn't have to understand name vs type to begin using state filtering.

**Comment by bunchesofdonald** [comment]
Thanks @madkinsz that's updated.

File: src/prefect/cli/flow_run.py [hunk 1]
```diff
@@ -44,21 +45,27 @@ async def inspect(id: UUID):
...
 async def ls(
-    flow_name: List[str] = None,
-    limit: int = 15,
-    state_type: List[StateType] = None,
+    flow_name: List[str] = typer.Option(None, help="Name of the Flow"),
+    limit: int = typer.Option(15, help="Maximum number of Flow Run's to list"),
+    state: List[str] = typer.Option(None, help="Name of the Flow Run's state"),
+    state_type: List[StateType] = typer.Option(
+        None, help="Type of the Flow Run's state"
+    ),
```
Comment (madkinsz) [inline src/prefect/cli/flow_run.py:53]: "Flow" and "Flow Run" should not be capitalized in accordance with our docs styling.

File: tests/cli/test_flow_run.py [hunk 2]
```diff
@@ -16,6 +28,51 @@ async def assert_flow_run_is_deleted(orion_client, flow_run_id: UUID):
         await orion_client.read_flow_run(flow_run_id)
 
 
+def assert_flow_runs_in_result(result, expected, unexpected=None):
+    output = result.stdout.strip()
+
+    # When running in tests the output of the columns in the table are
+    # truncated, so this only asserts that the first 20 characters of each
+    # flow run's id is in the output.
```
Comment (madkinsz) [inline tests/cli/test_flow_run.py:36]: We shouldn't ever be truncating ids in CLI output, right? Can we fix that? Could be a separate issue.

Comment (madkinsz) [reply tests/cli/test_flow_run.py:36]: Hm. We disable soft-wrap during test runs https://rich.readthedocs.io/en/stable/console.html#soft-wrapping which should prevent cropping. I'm fine with adding an issue to the backlog.

**Review by madkinsz — APPROVED** [review]

File: src/prefect/cli/flow_run.py [hunk 3]
```diff
@@ -44,21 +45,27 @@ async def inspect(id: UUID):
 
 @flow_run_app.command()
 async def ls(
-    flow_name: List[str] = None,
-    limit: int = 15,
-    state_type: List[StateType] = None,
+    flow_name: List[str] = typer.Option(None, help="Name of the flow"),
+    limit: int = typer.Option(15, help="Maximum number of flow run's to list"),
```
Comment (madkinsz) [inline src/prefect/cli/flow_run.py:49]:
```suggestion
    limit: int = typer.Option(15, help="Maximum number of flow runs to list"),
```

File: tests/cli/test_flow_run.py (hunk 2 above)
Comment (bunchesofdonald) [reply tests/cli/test_flow_run.py:36]: https://github.com/PrefectHQ/prefect/issues/6311

**Pull request merged**